spec/lib/rbac/filterer_spec.rb

```diff
@@ -830,6 +830,18 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
             expect(objects).to eq(targets)
           end
 
+          it "returns the correct results when targets are of different types" do
+            @ems3 = FactoryGirl.create(:ems_vmware, :name => 'ems3')
+            @ems4 = FactoryGirl.create(:ems_microsoft, :name => 'ems4')
+
+            targets = [@ems2, @ems4, @ems3, @ems]
+
+            results = described_class.search(:targets => targets, :user => user)
+            objects = results.first
+            expect(objects.length).to eq(4)
+            expect(objects).to eq(targets)
```

Comment:
Ugh, are these guaranteed to be in the same order?  Maybe we should use `contain_exactly` or it's similar form `match_array`